libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/store/AWSCredentialsRefresher.java

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright 2020 Verizon Media
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.yahoo.athenz.common.server.store;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.yahoo.athenz.common.server.util.ConfigProperties;
+import com.yahoo.athenz.zms.ResourceException;
+import org.eclipse.jetty.util.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.yahoo.athenz.common.ServerCommonConsts.*;
+
+public class AWSCredentialsRefresher implements Closeable {
```

Comment:
Do we really need this class nowadays? Doesn't AWS provide us with a credentials provider that is asynchronously refreshed in the background since we're running in EC2 thus we don't need to write this class at all?

Reply:
Yes, we usually don't need to specify credentials when working in AWS but I assumed that there is a reason for us getting the credentials through the metadata in CloudStore (that I'm not familiar with).
If it is just for historical reasons (wasn't possible in older AWS SDK), I'll just leave the DataStore as is and just build the s3client without credentials

Reply:
Yes, that's the right approach. I think we had some issues (this is several years back now) with the credentials provider so we opted to fetch the creds ourselves and it's been on my list all this time to revisit this code. But let's just remove it and rely on providers from aws directly. At some point we also need to look into what it would take to move to 2.x but for now let's stick with 1.x of aws java sdk.